### ISSUE TITLE
`ObjectMapperTest.testCopyWith()` failing on `2.16` branch on system with non-unix system linefeed

### ISSUE THREAD

**Issue by garretwilson** [FasterXML/jackson-databind#4168]
### Search before asking

- [X] I searched in the [issues](https://github.com/FasterXML/jackson-databind/issues) and found nothing similar.

### Describe the bug

When I try to build on the `2.16` branch using `mvn clean verify`, I get:

```
[ERROR] Failures: 
[ERROR]   ObjectMapperTest.testCopyWith:206 expected:<{[
  "color" : "Black",
  "free" : "true",
  "pages" : "204.04"]
}> but was:<{[
  "color" : "Black",
  "free" : "true",
] "pages" : "204.04"
}>
```

Glancing at the source code, I see it's comparing against a string with embedded `\n` (`LF`) literals.

```java
String readResultAsString = "{\n  \"color\" : \"Black\",\n  \"free\" : \"true\",\n  \"pages\" : \"204.04\"\n}";
assertEquals(readResultAsString, mapper.writeValueAsString(readResult));
```

Just on the face of it (assuming normally the developers wouldn't knowingly commit breaking code), I would bet you a cookie that Jackson is producing output using the system-configured line ending (which may in fact be appropriate), yet the person who wrote the test assumed that Jackson is producing `LF` line endings. The test passed for the developer because they are using a *nix machine. In my case, I'm on a Windows machine so perhaps it produced `CRLF` endings. (I'm just guessing by glancing at the code.)

If that is indeed the problem, you'll want to either use `System.lineSeparator()` when constructing the expected output; or (perhaps a better approach) create a separate generalized output comparison method that normalizes line endings to be used across all the tests (except the few methods, if any, specifically checking for the line endings).

### Reproduction

<-- Any of the following
1. Brief code sample/snippet: include here in preformatted/code section
2. Longer example stored somewhere else (diff repo, snippet), add a link
3. Textual explanation: include here
 -->
```java
// Your code here
```

**Comment by JooHyukKim** [comment]
Strange, builds on mine and CI passes. Could you check your mvn version via `mvn --version`? Should be java 8.

**Comment by garretwilson** [comment]
> Strange, builds on mine and CI passes.

Are you running on a Windows or Linux machine?

**Comment by garretwilson** [comment]
> Sorry, I missed the windows part.

Yeah, read the description closely. I pretty much already know what the problem is, and I explain how to fix it. It's just a guess from reading the code. If I'm wrong I'll owe you a cookie. 😆

**Comment by JooHyukKim** [comment]
No need for cookies 😂, thanks tho!

**Comment by garretwilson** [comment]
Note that if you go the route of using the default system line endings in the string being compared, there are 100 ways to do it, but one trick is to use a variation of `String.form(…)` and embed `%n`. Then formatting will convert that to the line ending configured for the system.

**Comment by JooHyukKim** [comment]
How about title be something like this?

    Test failure on Non *-nix Systems (e.g. Windows), aroundLine Endings `ObjectMapperTest.testCopyWith()`

**Comment by garretwilson** [comment]
> How about title be something like this?

Sure. I didn't test on a *nix machine, so my assessment of the problem was just an educated guess. If you found that this was indeed the problem, feel free to tweak the title as you feel appropriate.

**Comment by JooHyukKim** [comment]
> feel free to tweak the title as you feel appropriate.

Regarding that, I don't have the authorization 😅. I am just a volunteer (tho Github says "member") enjoying working with Jackson team. Sorry for requesting a bunch.

**Comment by garretwilson** [comment]
Oh, interesting. I didn't know you didn't have permissions to change the title. OK, I'll update. No problem.

**Comment by garretwilson** [comment]
Can you just confirm that you have verified that line endings are indeed the source of the problem, before I update the title? Because I haven't verified this personally.

**Comment by JooHyukKim** [comment]
> Can you just confirm that you have verified that line endings are indeed the source of the problem, 

Sorry I am not able to. I am omw to work to dig deeper. But note that we might need to check https://github.com/FasterXML/jackson-core where `PrettyPrinter` comes from.

**Comment by garretwilson** [comment]
In that case I feel it is premature to update the title to reflect a cause we have not verified, and that is based only on my educated guess after glancing at the code for literally five seconds or so. Currently the title is accurate based upon our knowledge: there is a certain test that breaks under certain circumstances on the `2.16` branch.

**Comment by cowtowncoder** [comment]
Definitely sounds like system linefeed issue wrt testing. Would be great to have sanity check tests for CI on Windows system; I assume Github has runners for this but I have not done it before.
I don't have a windows machine myself to test it with but I'll see what I can do here.

**Comment by JooHyukKim** [comment]
> Would be great to have sanity check tests for CI on Windows system; I assume Github has runners for this but I have not done it before.

+1 on Windows runner. There is windows machine that can be configured via `runs-on: windows-latest`. If no one started, by the time I get off work, will try to work on it 👍🏼

**Comment by cowtowncoder** [comment]
Ok, I added a work-around that hopefully resolves the immediate issue. But having test suite run on windows too would be useful. Not sure if that requires separate workflow or just different job, but either way would be useful to weed out unintentional platform deps.

**Comment by pjfanning** [comment]
In GitHub Actions yml, you can set `os` to a value like `windows-latest`. You can even define an array with a list of `os` values. See https://github.com/FasterXML/jackson-databind/blob/2.16/.github/workflows/main.yml#L29

But, the commands we use in `main.yml` are *Nix oriented (eg `/.mvnw`). So it would probably be better to have a separate win.yml that is similar to main.yml but windows specific.

**Comment by JooHyukKim** [comment]
I see #4172 going on 🚀

**Comment by pjfanning** [comment]
https://github.com/FasterXML/jackson-databind/pull/4172 is working now

**Comment by cowtowncoder** [comment]
I think CI tests now pass on Windows, as per my fix #4170 and verification via #4172, closing.
